sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java

```diff
@@ -304,22 +299,23 @@ private void closeClient() {
     /*
      * This method renews the ownership of currently owned partitions
      */
-    private void renewOwnership(Map<String, PartitionOwnership> partitionOwnershipMap) {
+    private Mono<Void> renewOwnership(Map<String, PartitionOwnership> partitionOwnershipMap) {
         morePartitionsToClaim.set(false);
         // renew ownership of already owned partitions
-        checkpointStore.claimOwnership(partitionPumpManager.getPartitionPumps().keySet()
+        return checkpointStore.claimOwnership(partitionPumpManager.getPartitionPumps().keySet()
             .stream()
             .filter(
                 partitionId -> partitionOwnershipMap.containsKey(partitionId) && partitionOwnershipMap.get(partitionId)
                     .getOwnerId().equals(this.ownerId))
             .map(partitionId -> createPartitionOwnershipRequest(partitionOwnershipMap, partitionId))
             .collect(Collectors.toList()))
-            .subscribe(partitionPumpManager::verifyPartitionConnection,
```

Comment:
One thing need considering is error handling.

On "subscribe" with Runnable, if this "renewOwnership" part get error, I think it won't affect the loadBalance Mono.

However, if it is now on this Mono that get returned from loadBalance, now any error will affect this loadBalance Mono.

Reply:
Yes, you are right! I may need to hide `renewOwnership` error from being propagated to keep them consistent. Let me double check the logic.

Reply:
After checked,  `renewOwnership` will have the same behavior after I changed. Because the loadbalance will call `renewOwnership` once as the repeat() condition is not met. 

But for `claimOwnership`, as there are multiple threads are running to claim, if one thread met error, it won't affect other threads. However, it could be a issue if the error is not transient, the load balance will do the claim ownership forever. So I think we need to distinguish the different error. If error is transient, we could continue by using onErroResume. if not, we should stop the loadBalance Mono.

As we return `Mono.empty()` for all errors in claim onwership in checkpoint, currently I think we can stop this loadbalance Mono if we catch a error and load balance can still be triggered for next iteration.

Reply:
But some error handling steps (logging and set false) will be duplicated after changed, let me see how to clean up them.